packages/main/src/components/VariantManagement/index.tsx

```diff
@@ -318,7 +318,8 @@ const VariantManagement = forwardRef<HTMLDivElement, VariantManagementPropTypes>
   }, [safeChildrenWithFavorites]);
 
   // todo: this applies if `readOnly` is set to `false` as well since the value is read via data-attribute and is therefore a string not a boolean
-  const showSaveBtn = dirtyState && !Object.prototype.hasOwnProperty.call(selectedVariant, 'readOnly');
+  const showSaveBtn =
+    dirtyState && selectedVariant && !Object.prototype.hasOwnProperty.call(selectedVariant, 'readOnly');
```

Comment:
It turns out that there's another issue with this approach. The `hasOwnProperty` check is only there to workaround React 19 (previously) handling data-attributes differently then React 18 and below. At least this was the case in the release candidate version of React we used when implementing this.
Unfortunately there's a big oversight on our end here, because with this implementation the save button is displayed only if `readOnly` isn't set at all.

Could you please refactor it like this?

```suggestion
  const showSaveBtn =
    dirtyState && selectedVariant && !selectedVariant.readOnly
```

I'll take a deeper look into how React19 treats data-attributes in the latest rc and also improve the test of the `readOnly` behavior in a later PR.